src/pipecat/services/ai_services.py

```diff
@@ -393,15 +393,18 @@ async def _stop_words_task(self):
             self._words_task = None
 
     async def _words_task_handler(self):
+        last_pts = 0
         while True:
             try:
                 (word, timestamp) = await self._words_queue.get()
                 if word == "LLMFullResponseEndFrame" and timestamp == 0:
-                    await self.push_frame(LLMFullResponseEndFrame())
+                    frame = LLMFullResponseEndFrame()
+                    frame.pts = last_pts
```

Comment:
Before this frame would be added after all the audio and since it didn't have a `pts` it would just be processed after the audio is spoken which should coincide with the text timestamps. But to make it more accurate we make sure it has a pts and its processed just after the text.